scripts/linters/pylint_extensions.py

```diff
@@ -1469,6 +1475,80 @@ def process_module(self, node):
                     line=closing_line_index_of_fileoverview + 1)
 
 
+class NewlineBelowClassDocstring(checkers.BaseChecker):
+    """Checks if there is a single newline below the class docstring."""
+
+    __implements__ = interfaces.IRawChecker
+    name = 'newline-below-class-docstring'
+    priority = -1
+    msgs = {
+        'C0026': (
+            'No single newline below class docstring.',
+            'newline-below-class-docstring',
+            'Please add a single newline below class docstring.'
+        )
+    }
+
+    def process_module(self, node):
+        """Process a module to ensure that there is a single newline below
+        class docstring.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node to access module content.
+        """
+
+        in_class = False
+        in_docstring = False
+        class_has_docstring = False
+        file_content = read_from_node(node)
+        file_length = len(file_content)
+        blank_line_counter = 0
+        prev_line = ''
+
+        for line_num in python_utils.RANGE(file_length):
+            line = file_content[line_num].strip()
+
+            if line_num > 0:
+                prev_line = file_content[line_num - 1].strip()
+
+            # Check if it is a docstring and not some multi-line string.
+            if prev_line.startswith(b'class ') or in_class:
+                in_class = True
+                if prev_line.endswith(b'):'):
+                    in_class = False
+                    if line.startswith(b'"""'):
+                        in_docstring = True
+                        class_has_docstring = True
+
+            if line.endswith(b'"""'):
+                in_docstring = False
+                in_class = False
+                continue
+
+            if in_docstring:
+                continue
+
+            if class_has_docstring:
+                if file_content[line_num] == b'\n':
+                    blank_line_counter += 1
+
+                elif line.startswith(('class ', 'def', '@')):
+                    blank_line_counter = 0
+                    class_has_docstring = False
```

Comment:
Can youplease explain this change? Also, there can be a constant defined just after the class defination, are we considering that case? 

Currently I'm unable to understand this function can you please add a code comment before every chck and I would suggest you to make it more readable.

Reply:
Done.

Reply:
> Can you please explain this change? Also, there can be a constant defined just after the class definition, are we considering that case

Yes we are considering that case also